### PR TITLE
Add support for regional secret version resource `google_secret_manager_regional_secret_version`

### PR DESCRIPTION
**Pull request by abheda-crest**
Add support for new regional secret version resource `google_secret_manager_regional_secret_version`.
More info about regional secrets: https://cloud.google.com/secret-manager/docs/regional-secrets-overview

A point to be noted is that while importing the `google_secret_manager_regional_secret_version` resources having the `is_secret_data_base64` field set to true, the `secret_data` is decoded from the base64-encoded string received from API. It is potentially due to condition in the flattener for payload (which is in place for the sake of `is_secret_data_base64` field to work). It seems the import is working fine when the `is_secret_data_base64` is not provided or set to false. While doing a terraform import, there is no way to determine if the resource to be imported requires decoding of base64 response.
References: https://github.com/hashicorp/terraform-provider-google/issues/10129 & https://github.com/GoogleCloudPlatform/magic-modules/pull/8873

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_secret_manager_regional_secret_version`
```

### CODE REVIEW

**Review by gptSanyam — CHANGES_REQUESTED** [review]

File: mmv1/products/secretmanagerregional/RegionalSecretVersion.yaml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    name: secret
+    url_param_only: true
+    resource: RegionalSecret
+    imports: name
+    required: true
+    immutable: true
+    description: |
+      Secret Manager regional secret resource.
+  - !ruby/object:Api::Type::String
+    name: location
```
[the remark this answers is not in the document]
Comment (abheda-crest) [reply mmv1/products/secretmanagerregional/RegionalSecretVersion.yaml:100]: The base_url as seen in the [Product.yaml](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/secretmanagerregional/product.yaml) contains `{{location}}` parameter which is replaced using the [`ReplaceVars` function](https://github.com/hashicorp/terraform-provider-google/blob/main/google/services/secretmanagerregional/resource_secret_manager_regional_secret.go#L298) before making any API call. For this, we require the location field to be set in the terraform state. Hence, we have kept the location parameter as computed. We infer the location field using the pre_create and pre_read custom code and override the url to include the information about the location parameter in the base_url.

File: mmv1/products/secretmanagerregional/RegionalSecretVersion.yaml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      secret_id: 'secret-version'
+      data: 'secret-data'
+    ignore_read_extra:
+      - 'deletion_policy'
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  post_create: templates/terraform/post_create/regional_secret_version.go.erb
+  decoder: templates/terraform/decoders/treat_destroyed_state_as_gone.erb
+  pre_delete: templates/terraform/pre_delete/regional_secret_version_deletion_policy.go.erb
+  pre_read: templates/terraform/pre_read/secret_manager_regional_secret_version.go.erb
+  pre_create: templates/terraform/pre_create/secret_manager_regional_secret_version.go.erb
```
[the remark this answers is not in the document]
Comment (abheda-crest) [reply mmv1/products/secretmanagerregional/RegionalSecretVersion.yaml:71]: We infer the location field from the `secret` field using the [pre_create](https://github.com/GoogleCloudPlatform/magic-modules/pull/11699/files/2494b6ef497ee49c3f2cbf877d3117c7191ae9a8#diff-4c361103de354f5b3067ac42dff90b1cf5e490291ff51082a56ef94e869dc4df) and [pre_read](https://github.com/GoogleCloudPlatform/magic-modules/pull/11699/files/2494b6ef497ee49c3f2cbf877d3117c7191ae9a8#diff-8326038022f4cba63d1058d94b9d6f347740cb2e9ae73b57b52a6385b16ac86cR15-R31) custom code and override the url to include the information about the location parameter in the base_url.

File: mmv1/templates/terraform/post_create/regional_secret_version.go.erb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+name, ok := res["name"]
+if !ok {
+	return fmt.Errorf("Create response didn't contain critical fields. Create may not have succeeded.")
+}
+if err := d.Set("name", name.(string)); err != nil {
+	return fmt.Errorf("Error setting name: %s", err)
+}
+d.SetId(name.(string))
+
+_, err = expandSecretManagerRegionalRegionalSecretVersionEnabled(d.Get("enabled"), d, config)
```
[the remark this answers is not in the document]
Comment (abheda-crest) [reply mmv1/templates/terraform/post_create/regional_secret_version.go.erb:25]: This function is defined in the [custom_expand](https://github.com/GoogleCloudPlatform/magic-modules/pull/11699/files/2494b6ef497ee49c3f2cbf877d3117c7191ae9a8#diff-521c8c92b4d372df8c386808bc2660bc25cd31a2ca79334e5a0028b2b7a7db5f) custom code for the [enabled field](https://github.com/GoogleCloudPlatform/magic-modules/pull/11699/files/2494b6ef497ee49c3f2cbf877d3117c7191ae9a8#diff-3ef207c5c2c04b2b56a3627d812a2755e025d3a23fec9794c80fb1aeb334f9edR146) as we need to explicitly make the enable/disable call based on the value of the `enabled` field.

File: mmv1/templates/terraform/custom_import/regional_secret_version.go.erb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	if err := tpgresource.ParseImportId([]string{"(?P<name>.+)"}, d, config); err != nil {
+		return nil, err
+	}
+
+	name := d.Get("name").(string)
+	secretRegex := regexp.MustCompile("(projects/.+/locations/.+/secrets/.+)/versions/.+$")
+	versionRegex := regexp.MustCompile("projects/(.+)/locations/(.+)/secrets/(.+)/versions/(.+)$")
+
+	parts := secretRegex.FindStringSubmatch(name)
+	if len(parts) != 2 {
```
[the remark this answers is not in the document]
Comment (abheda-crest) [reply mmv1/templates/terraform/custom_import/regional_secret_version.go.erb:27]: As mentioned above, we need the location field for the base_url of the regional APIs hence we do it for `pre_create`, `pre_read` and `custom_import` functionalities.

File: mmv1/products/secretmanagerregional/RegionalSecretVersion.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (gptSanyam) [reply mmv1/products/secretmanagerregional/RegionalSecretVersion.yaml:100]: What happens if a customer puts a value of location while creating secret version which is different from the one defined in secret?
We will override in pre_create and pre_read and the request goes to a different endpoint.
Such request with difference in location should ideally be rejected at the least. Can we introduce such a check if not done already?

File: mmv1/products/secretmanagerregional/RegionalSecretVersion.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (gptSanyam) [reply mmv1/products/secretmanagerregional/RegionalSecretVersion.yaml:71]: Understood.

File: mmv1/templates/terraform/custom_import/regional_secret_version.go.erb (hunk 4 above)
[the remark this answers is not in the document]
Comment (gptSanyam) [reply mmv1/templates/terraform/custom_import/regional_secret_version.go.erb:27]: Understood. Thanks.

File: mmv1/products/secretmanagerregional/RegionalSecretVersion.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (abheda-crest) [reply mmv1/products/secretmanagerregional/RegionalSecretVersion.yaml:100]: Customer won't be able to provide the location field as it is [output only field](https://github.com/GoogleCloudPlatform/magic-modules/pull/11699/files/2494b6ef497ee49c3f2cbf877d3117c7191ae9a8#diff-3ef207c5c2c04b2b56a3627d812a2755e025d3a23fec9794c80fb1aeb334f9edR102) from terraform side (i.e. Computed field). We will infer the location based on the regional secret resource reference.

[the remark this answers is not in the document]
Comment (gptSanyam) [reply mmv1/products/secretmanagerregional/RegionalSecretVersion.yaml:100]: Aah! missed the output = True bit.
Thanks for clarification.

**Review by gptSanyam — APPROVED** [review]
LGTM

File: mmv1/third_party/terraform/services/secretmanagerregional/resource_secret_manager_regional_secret_version_test.go.erb [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+package secretmanagerregional_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccSecretManagerRegionalRegionalSecretVersion_import(t *testing.T) {
```
Comment (shuyama1) [inline mmv1/third_party/terraform/services/secretmanagerregional/resource_secret_manager_regional_secret_version_test.go.erb:11]: Out of curiosity, why do we need this test specifically, I don't see it's different from the first step of `TestAccSecretManagerRegionalRegionalSecretVersion_update`

**Review by shuyama1 — COMMENTED** [review]
This is a regional version of `google_secret_manager_secret_version`. Overall looks good to me. I only have a small question regarding a test case. Thanks!

File: mmv1/third_party/terraform/services/secretmanagerregional/resource_secret_manager_regional_secret_version_test.go.erb (hunk 5 above)
Comment (abheda-crest) [reply mmv1/third_party/terraform/services/secretmanagerregional/resource_secret_manager_regional_secret_version_test.go.erb]: I included this test case to maintain consistency with the other resources and to test a basic import scenario. However, I understand your point about the redundancy, so I've decided to remove this test case.

**Review by shuyama1 — APPROVED** [review]
Thank you!

**Pull request merged**